virtual-desktop/src/app/window-manager/mvd-window-manager/desktop/desktop.component.ts

```diff
@@ -155,52 +156,67 @@ export type DesktopTheme = {
     let desktop:ZLUX.Plugin = ZoweZLUX.pluginManager.getDesktopPlugin();
     let recognizersUri = ZoweZLUX.uriBroker.pluginConfigUri(desktop,'recognizers');
-    let actionsUri = ZoweZLUX.uriBroker.pluginConfigUri(desktop,'actions');
-    this.log.debug("ZWED5309I", recognizersUri, actionsUri); //this.log.debug(`Getting recognizers from "${recognizersUri}", actions from "${actionsUri}"`);
+    this.log.debug(`Getting recognizers from "${recognizersUri}"`);
     this.http.get(recognizersUri).subscribe((config: any)=> {
-      if (config) {
+      if (config && config.contents) {
         let appContents = config.contents;
-        let appsWithRecognizers:string[] = [];
         plugins.forEach((plugin:ZLUX.Plugin)=> {
-          let id = plugin.getIdentifier();
-          if (appContents[id]) {
-            appsWithRecognizers.push(id);
+          const id = plugin.getIdentifier();
+          if (appContents[id] && appContents[id].recognizers) { // If config has pre-existing recognizers for this plugin id,
+            appContents[id].recognizers.forEach((recognizerObject: any)=> {
+              ZoweZLUX.dispatcher.addRecognizerObject(recognizerObject); // register each object with the Dispatcher.
+            });
+            this.log.info(`ZWED5055I`, appContents[id].recognizers.length, id); //this.log.info(`Loaded ${appContents[id].recognizers.length} recognizers for App(${id})`);    
           }
         });
-        appsWithRecognizers.forEach(appWithRecognizer=> {
-          appContents[appWithRecognizer].recognizers.forEach((recognizerObject:ZLUX.RecognizerObject)=> {
-            ZoweZLUX.dispatcher.addRecognizerObject(recognizerObject);
-          });
-          this.log.info(`ZWED5055I`, appContents[appWithRecognizer].recognizers.length, appWithRecognizer); //this.log.info(`Loaded ${appContents[appWithRecognizer].recognizers.length} recognizers for App(${appWithRecognizer})`);
-        });
       }
     });
+  }
+
+  getAndDispatchActions(plugins: ZLUX.Plugin[]) {
+    let desktop:ZLUX.Plugin = ZoweZLUX.pluginManager.getDesktopPlugin();
+    let actionsUri = ZoweZLUX.uriBroker.pluginConfigUri(desktop,'actions');
+    this.log.debug(`Getting actions from "${actionsUri}"`);
     this.http.get(actionsUri).subscribe((config: any)=> {
-      if (config) {
+      if (config && config.contents) {
         let appContents = config.contents;
-        let appsWithActions:string[] = [];
         plugins.forEach((plugin:ZLUX.Plugin)=> {
-          let id = plugin.getIdentifier();
-          if (appContents[id]) {
-            appsWithActions.push(id);
+          const id = plugin.getIdentifier();
+          if (appContents[id] && appContents[id].actions) { // If config has pre-existing actions for this plugin id,
+            appContents[id].actions.forEach((actionObject: any)=> {
+              ZoweZLUX.dispatcher.addRecognizerObject(actionObject); // register each object with the Dispatcher.
```

Comment:
This looks like an accident of some kind (changing from registerAbstractAction to addRecognizerObject)